File: StudAid/StudAid.Model/Subject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StudAid.Model
{
    public class Subject
    {
        public int SubjectId { get; set; }
     
        public string SubjectName { get; set; }

        public int? CategoryId { get; set; }
    }
}

File: StudAidMobileApp/lib/models/subject.dart
import 'package:json_annotation/json_annotation.dart';

part 'subject.g.dart';

@JsonSerializable()
class Subject {
  int? subjectId;
  String? subjectName;
  int? categoryId;

  Subject() {}
  factory Subject.fromJson(Map<String, dynamic> json) =>
      _$SubjectFromJson(json);

  Map<String, dynamic> toJson() => _$SubjectToJson(this);
}
